### PR TITLE
Update light client specifications for Electra

### PR DESCRIPTION
**Pull request by etan-status**
Electra introduces two changes that affect light client data handling:

1. The `ExecutionPayloadHeader` is extended with new fields. This is handled similarly as before with the Deneb fork.

2. The `BeaconState` generalized indices change due to lack of EIP-6493. This is handled by making the generalized index be fork dependent via a helper function that computes it dynamically. Furthermore, the case where pre-Electra light client data is consumed by an Electra based `LightClientStore` requires normalizing the shorter proof of the pre-Electra data to fit into the Electra data structure by prepending a zero hash.

### CODE REVIEW

**Comment by etan-status** [comment]
[consensus-spec-tests-3811.zip](https://github.com/user-attachments/files/15925682/consensus-spec-tests-3811.zip)

**Comment by etan-status** [comment]
[consensus-spec-tests-3811.zip](https://github.com/user-attachments/files/15927975/consensus-spec-tests-3811.zip)

File: specs/electra/light-client/full-node.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        # [New in Electra:EIP6110:EIP7002:EIP7251]
+        if epoch >= ELECTRA_FORK_EPOCH:
+            execution_header.deposit_requests_root = hash_tree_root(payload.deposit_requests)
+            execution_header.withdrawal_requests_root = hash_tree_root(payload.withdrawal_requests)
+            execution_header.consolidation_requests_root = hash_tree_root(payload.consolidation_requests)
+
+        execution_branch = ExecutionBranch(
+            compute_merkle_proof(block.message.body, EXECUTION_PAYLOAD_GINDEX))
+    else:
+        # Note that during fork transitions, `finalized_header` may still point to earlier forks.
```
Comment (dapplion) [inline specs/electra/light-client/full-node.md:62]: So far in Ethereum networks we have aligned fork epochs to light-client sync periods. Under what circumstance could that happen then?

Comment (etan-status) [reply specs/electra/light-client/full-node.md:62]: The head advances into the new fork before the finalized cp does. there is a short transition period during which the finalized header has to be normalized into the new datatype.

**Review by hwwhww — APPROVED** [review]
lgtm. thank you @etan-status for your solid work again!

**Pull request merged**